### ISSUE TITLE
Crash in `CompilationSupport` when building `//ios-app:UrlGetClasses` on Linux

### ISSUE THREAD

**Issue by lberki** [bazelbuild/bazel#6741]
Reproduction on a Linux machine:

`bazel build --nobuild //ios-app:UrlGetClasses` 

Output with Bazel `0.19.2` (I'll omit the top of the stack trace since the important part is the one in the configured target factory):

```
Caused by: java.lang.NullPointerException
	at com.google.devtools.build.lib.rules.objc.CompilationSupport.getFeatureConfiguration(CompilationSupport.java:560)
	at com.google.devtools.build.lib.rules.objc.CompilationSupport.compile(CompilationSupport.java:318)
	at com.google.devtools.build.lib.rules.objc.CompilationSupport.ccCompileAndLink(CompilationSupport.java:395)
	at com.google.devtools.build.lib.rules.objc.CompilationSupport.registerCompileAndArchiveActions(CompilationSupport.java:1040)
	at com.google.devtools.build.lib.rules.objc.CompilationSupport.registerCompileAndArchiveActions(CompilationSupport.java:1089)
	at com.google.devtools.build.lib.rules.objc.CompilationSupport.registerCompileAndArchiveActions(CompilationSupport.java:984)
	at com.google.devtools.build.lib.rules.objc.ObjcLibrary.create(ObjcLibrary.java:86)
	at com.google.devtools.build.lib.rules.objc.ObjcLibrary.create(ObjcLibrary.java:44)
```

HEAD is the same, except for a different line number, but it's the same `XcodeConfig.getXcodeVersion(ruleContext)` that returns NULL. I suppose it's because the C++ toolchain is not an `apple_cc_toolchain` but a vanilla `cc_toolchain`?

**Comment by lberki** [comment]
I'm surprised our integration test battery did not catch this. Bazel@HEAD crashing on our examples is a pretty awful bug...

**Comment by irengrig** [comment]
It crashes when Blaze Intellij plugin tries to build Bazel (in more complicated case).
And it is not the plugin itself, it is Bazel 0.19.2.

**Comment by tetromino** [comment]
I tried it on Linux now, cannot reproduce the crash with bazel head (c3e2de37b24160a987762c8fad233d8ed4937ac0). At what commit did you see the crash? Can you still reproduce it?

**Comment by irengrig** [comment]
It crashes with Bazel 0.19.2 and sorry, I think I already updated the HEAD so I am not sure about that variant.

**Comment by tetromino** [comment]
Bisect shows that this was fixed on Monday by @mhlopko with 3aa807910cf31428ae40a99567682b09b94e5acc.

**Comment by irengrig** [comment]
Great! Thank you so much!
